lib/cocoapods-core/source/manager.rb

```diff
@@ -0,0 +1,410 @@
+module Pod
+  class Source
+    class Manager
+      # @return [Pathname] The directory that contains the source repo
+      #         directories.
+      #
+      attr_reader :repos_dir
+
+      def initialize(repos_dir)
+        @repos_dir = Pathname(repos_dir).expand_path
+      end
+
+      # @return [Array<Pathname>] The source repo directories.
+      #
+      def source_repos
+        return [] unless repos_dir.exist?
+        repos_dir.children.select(&:directory?).sort_by { |d| d.basename.to_s.downcase }
+      end
+
+      # @return [Source::Aggregate] The aggregate of all the sources with the
+      #         known Pods.
+      #
+      def aggregate
+        aggregate_with_repos(source_repos)
+      end
+
+      # @return [Source::Aggregate] The aggregate of the sources from repos.
+      #
+      # @param  [Dependency] dependency
+      #         The dependency for which to find or build the appropriate.
+      #         aggregate. If the dependency specifies a source podspec repo
+      #         then only that source will be used, otherwise all sources
+      #         will be used.
+      #
+      def aggregate_for_dependency(dependency)
+        if dependency.podspec_repo
+          source = source_with_url(dependency.podspec_repo)
+          raise StandardError, '[Bug] Failed to find known source with the URL ' \
+            "#{dependency.podspec_repo.inspect}" if source.nil?
+
+          aggregate_with_repos([source.repo])
+        else
+          aggregate
+        end
+      end
+
+      # @return [Array<Source>] The list of the sources with the given names.
+      #
+      # @param  [Array<#to_s>] names
+      #         The names of the sources.
+      #
+      def sources(names)
+        dirs = names.map { |name| source_dir(name) }
+        dirs.map { |repo| source_from_path(repo) }
+      end
+
+      # @return [Array<Source>] The list of all the sources known to this
+      #         installation of CocoaPods.
+      #
+      def all
+        aggregate.sources
+      end
+
+      # @return [Array<Source>] The CocoaPods Master Repo source.
+      #
+      def master
+        sources(['master']).select { |s| s.repo.directory? }
+      end
+
+      # @!group Master repo
+
+      # @return [Pathname] The path of the master repo.
+      #
+      def master_repo_dir
+        repos_dir + 'master'
```

Comment:
This is duplicating the behavior of https://github.com/CocoaPods/Core/pull/316/files#diff-0c2b4ee53dd36d1e844328e23dbfb878R332 could be sources_dir('master') instead?